admin/server/auth/handlers.go

```diff
@@ -134,13 +162,27 @@ func (a *Authenticator) authStart(w http.ResponseWriter, r *http.Request, signup
 		sess.Values[cookieFieldRedirect] = redirect
 	}
 
+	customDomainFlow := r.URL.Query().Get("custom_domain_flow")
```

Comment:
Maybe a small comment to explain:

```suggestion
	// If this is part of the custom domain login flow, save that info in the cookie since we need that info when handling the auth callback.
	customDomainFlow := r.URL.Query().Get("custom_domain_flow")
```